elasticapm/traces.py

```diff
@@ -55,7 +59,57 @@
 execution_context = init_execution_context()
 
 
-class Transaction(object):
+class ChildDuration(object):
+    __slots__ = ("obj", "_nesting_level", "_start", "_duration")
+
+    def __init__(self, obj):
+        self.obj = obj
+        self._nesting_level = AtomicNumber(0)
+        self._start = AtomicNumber(0.0)
+        self._duration = AtomicNumber(0.0)
+
+    def start(self, timestamp):
+        if self._nesting_level.inc() == 1:
```

Comment:
I wouldn't expect a lot of contention on a single transaction or span. I think a threading.Lock is appropriate here, to ensure nesting_level/duration are consistent.

Reply:
... and that kills `AtomicNumber` :D